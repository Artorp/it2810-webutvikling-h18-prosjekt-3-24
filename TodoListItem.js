import React from 'react';
import { StyleSheet } from 'react-native';
import { Button, Icon, Text, View } from 'native-base';

export default class TodoListItem extends React.Component {
  render() {
    return (
      <View style={styles.todoListItem}>
        <Text style={styles.itemText}>{this.props.name}</Text>
        <Button iconRight rounded small info style={styles.itemButton}>
          <Text>Fjern</Text>
          <Icon name="trash" />
        </Button>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  todoListItem: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  itemText: {
    width: '75%',
  },
  itemButton: {
    width: '25%',
  },
});
