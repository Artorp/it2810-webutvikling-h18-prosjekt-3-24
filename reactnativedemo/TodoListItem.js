import React from "react";
import { ScrollView, StyleSheet, Text, View } from "react-native";

export default class TodoListItem extends React.Component {
  render() {
    return <Text style={styles.text}>Gjøremål</Text>;
  }
}

const styles = StyleSheet.create({
  text: {
    color: "white"
  }
});
