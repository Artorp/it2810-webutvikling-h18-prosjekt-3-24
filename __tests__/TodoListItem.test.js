import React from 'react';
import renderer from 'react-test-renderer';
import TodoListItem from '../components/TodoListItem';

test('blir tegnet riktig', () => {
  const tree = renderer.create(<TodoListItem name="Testgjøremål" />).toJSON();
  expect(tree).toMatchSnapshot();
});
