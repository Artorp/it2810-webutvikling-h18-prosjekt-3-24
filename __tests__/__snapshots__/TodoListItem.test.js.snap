// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blir tegnet riktig 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "width": "75%",
        },
      ]
    }
    uppercase={false}
    virtual={undefined}
  >
    Testgjøremål
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#62B1F6",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 57,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "width": "25%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 14,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
      virtual={undefined}
    >
      Fjern
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
